pkg/morph/deploy/nns.go

```diff
@@ -226,3 +227,174 @@ func lookupNNSDomainRecord(inv *invoker.Invoker, nnsContract util.Uint160, domai
+			continue
+		}
+
+		setGroupInManifest(&prm.localManifest, prm.localNEF, prm.committeeGroupKey, prm.localAcc.ScriptHash())
+
+		var vub uint32
+
+		// we pre-check 'already updated' case via MakeCall in order to not potentially
+		// wait for previously sent transaction to be expired (condition below) and
+		// immediately succeed
+		tx, err := committeeActor.MakeCall(nnsOnChainState.Hash, methodUpdate,
+			bLocalNEF, jLocalManifest, extraUpdateArgs)
+		if err != nil {
+			if isErrContractAlreadyUpdated(err) {
+				// FIXME: we get here when on-chain version is >= than the local one. If equals,
+				// this case must be considered as failure since version must be changed if code
+				// changes (NEF checksum differs here according to condition above). If greater,
+				// we should try to update at all. For simplicity, such condition is considered
+				// as success by current procedure now. To catch it, we need to pre-read version
+				// of the local contract using local NEF.
+				prm.logger.Info("NNS contract has already been updated, skip")
+				return nil
+			}
+		} else {
+			if updateTxValidUntilBlock > 0 {
+				prm.logger.Info("transaction updating NNS contract was sent earlier, checking relevance...")
+
+				if cur := prm.monitor.currentHeight(); cur <= updateTxValidUntilBlock {
+					prm.logger.Info("previously sent transaction updating NNS contract may still be relevant, will wait for the outcome",
+						zap.Uint32("current height", cur), zap.Uint32("retry after height", updateTxValidUntilBlock))
+					continue
+				}
+
+				prm.logger.Info("previously sent transaction updating NNS contract expired without side-effect")
+			}
+
+			prm.logger.Info("sending new transaction updating NNS contract...")
+
+			_, _, vub, err = committeeActor.Notarize(tx, nil)
+		}
+		if err != nil {
+			lackOfGAS := isErrNotEnoughGAS(err)
+			// here lackOfGAS can become true even if error corresponds to insufficient
+			// funds for the main transaction, not Notary balance problem. Right now we
+			// cannot distinguish between these two cases. So, we naively believe that the
+			// problem is in the Notary balance, because according to used procedure we make
+			// Notary deposit when at least 50GAS is available (most likely enough for all
+			// transactions).
+			if !lackOfGAS {
+				if !isErrNotaryDepositExpires(err) {
+					prm.logger.Error("failed to send transaction deploying NNS contract, will try again later", zap.Error(err))
+					continue
+				}
+			}
+
+			// same approach with in-place deposit is going to be used in other functions.
+			// Consider replacement with background process (e.g. blockchainMonitor
+			// internal) which periodically checks Notary balance and updates it when, for
+			// example, balance goes lower than 20% of desired amount or expires soon. With
+			// this approach functions like current will not try to make a deposit, but
+			// simply wait until it becomes enough.
```

Comment:
lets discuss a bit there https://github.com/nspcc-dev/neofs-node/pull/2408#discussion_r1252269316
if implementation will be estimated pretty easy, we'll do it here. Otherwise separate issue